docs/releases.md

```diff
@@ -0,0 +1,91 @@
+# Configlet releases
+
+## How to create a configlet release
+
+1. Check that the repo state is such that creating a release PR makes sense.
+
+1. Run [`bin/bump_version.nim`][bump_version], which also prompts to create a release PR if you have [`gh`][gh] installed on your machine.
+
+1. Ideally, check that `configlet lint` and `configlet sync` do not produce unexpected output on any track - diffing the output from the previous release.
```

Comment:
It would be great to have a script for this (checking for any changes in output)